data-in-pipeline/app/models.py

```diff
@@ -1,19 +1,25 @@
-from typing import Generic, TypeVar
+import uuid
+from datetime import datetime
+from typing import Any, Generic, TypeVar
 
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
 
 ExtractedData = TypeVar("ExtractedData")
 
 
-class Extracted(BaseModel, Generic[ExtractedData]):
-    """Generic type for a source document.
-
-    We can inherit this later down the line and augment it with the more
-    specific type to build a more specific source document.
-    """
+class ExtractedEnvelope(BaseModel, Generic[ExtractedData]):
+    """Envelope for extracted data from any source document."""
 
     data: ExtractedData
-    source: str
+    envelope_id: str = Field(default_factory=lambda: str(uuid.uuid4()))
+    source_name: str
+    source_record_id: str
+    raw_payload: str
+    content_type: str = "application/json"
+    connector_version: str
+    run_id: str | None = None
+    extracted_at: datetime = Field(default_factory=datetime.utcnow)
+    metadata: dict[str, Any] = Field(default_factory=dict)
```

Comment:
We have quite a lot of loose metadata around our estate, it would be nice top not propagate this and set if explicitly on this type as it feels like that is what this type is e.g. a set of data wrapped in metadata.